torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,304 @@
+            dirname_year = filename_year.split(".")[0]
+            pathname = os.path.join(self.root, dirname_year, filename_year)
+            if os.path.exists(pathname):
+                exists.append(True)
+            else:
+                exists.append(False)
+
+        if all(exists):
+            return
+
+        # Check if the zip files have already been downloaded
+        exists = []
+        for year in self.years:
+            pathname = os.path.join(
+                self.root, self.zipfile_glob.replace("*", str(year))
+            )
+            if os.path.exists(pathname):
+                exists.append(True)
+                self._extract()
+            else:
+                exists.append(False)
+
+        if all(exists):
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        for year in self.years:
+            download_url(
+                self.url.format(year),
+                self.root,
+                md5=self.md5s[year] if self.checksum else None,
+            )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        for year in self.years:
+            zipfile_name = self.zipfile_glob.replace("*", str(year))
+            pathname = os.path.join(self.root, zipfile_name)
+            extract_archive(
+                pathname, os.path.join(self.root, zipfile_name.split(".")[0])
+            )
+
+    def plot(
+        self,
+        sample: Dict[str, Any],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`RasterDataset.__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        mask = sample["mask"].squeeze().numpy()
+        ncols = 1
+
+        cmap: "np.typing.NDArray[np.int_]" = np.zeros((max(self.cmap) + 1, 4), np.int_)
+        for idx, cmap_val in self.cmap.items():
+            cmap[idx, :] = cmap_val
+
+        mask = cmap[mask]
+
+        showing_predictions = "prediction" in sample
+        if showing_predictions:
+            pred = sample["prediction"].squeeze().numpy()
+            pred = cmap[pred]
+            ncols = 2
+
+        fig, axs = plt.subplots(nrows=1, ncols=ncols, figsize=(ncols * 4, 4))
```

Comment:
If you use `squeeze=False`, it will prevent `axs` from changing shape based on the value of `ncols`, making it easier to index and making the below code simpler.